internal/controller/authproxyworkload_controller.go

```diff
@@ -276,6 +276,12 @@ func (r *AuthProxyWorkloadReconciler) needsAnnotationUpdate(wl workload.Workload
 		return false
 	}
 
+	// The user has set "None" as the rollout strategy. Ignore it.
+	if resource.Spec.AuthProxyContainer != nil &&
```

Comment:
This check is pretty simple, but since we're repeating it and it's a pretty important check, should we pull it out into a re-used function?

Reply:
Good idea. Fixed.